pytorch_lightning/trainer/trainer.py

```diff
@@ -1769,26 +1769,35 @@ def reset_train_dataloader(self, model: Optional["pl.LightningModule"] = None) -
             self.train_dataloader = self._data_connector._resolve_overfit_batches(self.train_dataloader)
 
         # automatically add samplers
-        self.train_dataloader = apply_to_collection(
-            self.train_dataloader,
-            DataLoader,
-            self._data_connector._prepare_dataloader,
-            shuffle=True,
-            mode=RunningStage.TRAINING,
-        )
+        if isinstance(self.train_dataloader, CombinedLoader):
+            self.train_dataloader = self._data_connector._prepare_dataloader(
+                self.train_dataloader, shuffle=True, mode=RunningStage.TRAINING
+            )
+            is_combined_loader = True
+        else:
+            self.train_dataloader = apply_to_collection(
+                self.train_dataloader,
+                DataLoader,
+                self._data_connector._prepare_dataloader,
+                shuffle=True,
+                mode=RunningStage.TRAINING,
+            )
+            is_combined_loader = False
+        loaders = self.train_dataloader.loaders if is_combined_loader else self.train_dataloader
```

Comment:
```suggestion
        self.train_dataloader = apply_to_collection(
            self.train_dataloader,
            DataLoader,
            self._data_connector._prepare_dataloader,
            shuffle=True,
            mode=RunningStage.TRAINING,
        )
        loaders = self.train_dataloader.loaders if isinstance(self.train_dataloader, CombinedLoader) else self.train_dataloader
```

Reply:
_prepare_dataloader can handle combined loaders

Reply:
I tried your suggestion, but it breaks the tests, DDP is then not handled correctly for the CombinedLoader.

Reply:
got it to work with a small change, thanks!

Reply:
ah! my bad. forgot to add that